console/views/store.py

```diff
@@ -65,62 +65,64 @@ def get_filtered_responses(logger, valid, tx_id, ru_ref, survey_id, datetime_ear
     return filtered_data
 
 
-def reprocess_transaction(logger, json_data):
-    logger.info("Reprocessing transaction", tx_id=json_data["tx_id"])
-    if json_data.get("invalid"):
-        del json_data["invalid"]
-    validate_response = send_data(
-        logger=logger, url=settings.SDX_VALIDATE_URL, json=json_data, request_type="POST")
-    if validate_response != 200:
-        json_data['invalid'] = "True"
-    send_data(logger=logger, url=settings.SDX_STORE_URL, json=json_data, request_type="POST")
+def reprocess(tx_id):
+    logger.debug('Reprocess function')
+    publisher = QueuePublisher(
+        settings.RABBIT_URLS,
+        'sdx-survey-notification-durable'
+    )
+
+    publisher.publish_message(tx_id, headers={'tx_id': tx_id})
+
+
+@store_bp.route('/reprocess', strict_slashes=False, methods=['POST'])
+@flask_security.login_required
+def reprocess_submission():
+    tx_id = str(request.get_data())[1:]
+    tx_id = tx_id.replace("'", "")
+    logger.debug('Begin reprocessing tx_id: {}'.format(tx_id))
+
+    reprocess(tx_id)
+
+    return "Data reprocessed", 200
 
 
 @store_bp.route('/store', strict_slashes=False, defaults={'page': 0}, methods=['GET', 'POST'])
 @store_bp.route('/store/<page>', strict_slashes=False, methods=['GET', 'POST'])
 @flask_security.login_required
 def store(page):
     audited_logger = logger.bind(user=flask_security.core.current_user.email)
+
+    valid = request.args.get('valid', type=str, default='')
```

Comment:
Did this need to move?

Seems a little odd to do this if we're dealing with a post as request.args are query string parameters

Reply:
This is where we thought there was an issue with the post request, if we remove it then it breaks

Reply:
I think we need to get to the bottom of this before we merge . Hopefully I will get a chance to look again later